### DISCUSSION TITLE
Fedimint crate naming scheme and directory structure

### DISCUSSION THREAD

**Discussion by elsirion** [fedimint/fedimint#2025] [General]
Currently we have at least two naming schemes for directories: `/crypto/derive-secret` contains the crate `fedimint-derive-secret` while `/fedimint-core` contains the crate `fedimint-core`. I wonder if we should a) always or b) never prefix with fedimint in the directory structure.

If we were to decide against redundant prefixing I'd also propose the following directory structure for modules `/modules/{dummy,ln,mint,wallet}/{core,server,client}`.

Deduplicating naming of the crates themselves seems out of the question imo since crates.io doesn't support hierarchies.

**Comment by dpc** [comment]
`/modules/{dummy,ln,mint,wallet}/{core,server,client}` seems like a win.

But while at it, maybe we should move all code crates under same subdir? Either `/fedimint/`, `/fm/` or `/pkgs/` ?

So

```
fm/modules/{dummy,ln,mint,wallet}/{core,server,client}
fm/{core,server}
fm/crypto/{tbs,aead}
```

?

**Reply by elsirion** [reply]
Sounds good to me, just need to coordinate timing so rebases don't kill us.
